src/generateBreaks/normalization.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+function getFieldValues (features, field) {
+  return features.map((feature, index) => {
+    const properties = feature.properties
+    const key = Object.keys(properties).filter(property => { return property === field })
+    const value = Number(properties[key])
+    if (isNaN(value)) throw new TypeError('Cannot use values from non-recognized or non-numeric field')
+    return value
+  })
+}
+
+function normalizeValues (values, features, classification) {
+  const normType = classification.normalizationType
+  switch (normType) {
+    case 'esriNormalizeByField': return normalizeByField(values, features, classification)
+    case 'esriNormalizeByLog': return normalizeByLog(values)
+    case 'esriNormalizeByPercentOfTotal': return normalizeByPercent(values)
+    default: throw new Error('Normalization not supported: ', normType)
+  }
+}
+
+function normalizeByField (values, features, classification) {
+  const normField = classification.normalizationField
+  if (normField) {
+    const normValues = getFieldValues(features, normField)
+    if (Array.isArray(normValues)) {
```

Comment:
this function is tough to read, should probably be broken into a few smaller ones